storage/remote/queue_manager.go

```diff
@@ -266,6 +272,48 @@ func (t *QueueManager) Stop() {
 	level.Info(t.logger).Log("msg", "Remote storage stopped.")
 }
 
+// Keep track of which series we know about for lookups when sending samples to remote.
+func (t *QueueManager) storeSeries(series []tsdb.RefSeries) {
+	for i := 0; i < len(series); i++ {
+		var ls = make(model.LabelSet)
+		for _, label := range series[i].Labels {
+			ls[model.LabelName(label.Name)] = model.LabelValue(label.Value)
+		}
+		_, ok := t.series[series[i].Ref]
+		if !ok {
```

Comment:
Under what circumstances will we find a label set already in the cache?

And can you move this above the building of the `model.LabelSet` please?

Finally, generally in go we aim for the least nesting possible; so we favour:

```golang
if _, ok := t.series[series[i].Ref]; ok {
  continue
}
```

Reply:
Previously I thought series records were repeated per segment file, but I was wrong. I can remove the check for now, I don't think there's a case where we would receive another series record for a reference ID that's already in the cache.